vulnerable.cpp

```
@@ -1 +1,108 @@
+        ifstream infile("/nonexistent_file.txt");
+        infile.getline(inputCommand, 50);  // 讀取超過 inputCommand 陣列大小
+
+        // 將長字串複製到較短的陣列（緩衝區溢位）
+        char processedCommand[10] = {0};
+        char longInput[] = "A very long input that exceeds buffer limit";
+        strcpy(processedCommand, longInput);
+        cout << "Input: " << inputCommand << ", Processed: " << processedCommand << endl;
+
+        // 動態記憶體配置，但未初始化內容即直接使用
+        int *data = new int[5];
+        cout << "Data[2] (可能未初始化): " << data[2] << endl;
```

Cppcheck (2.10-2)

[error] Memory is allocated but not initialized

(uninitdata)